api/v1alpha1/broom_types.go

```diff
@@ -17,19 +17,79 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"fmt"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+type BroomTarget struct {
+	Name      string            `json:"name,omitempty"`
+	Labels    map[string]string `json:"labels,omitempty"`
+	Namespace string            `json:"namespace,omitempty"`
+}
+
+type BroomAdjustmentType string
+
+const (
+	AddAdjustment BroomAdjustmentType = "Add"
+	MulAdjustment BroomAdjustmentType = "Mul"
+)
+
+type BroomAdjustment struct {
+	Type  BroomAdjustmentType `json:"type"`
+	Value string              `json:"value"`
+}
+
+func (adj BroomAdjustment) IncreaseMemory(m *resource.Quantity) error {
+	switch adj.Type {
+	case AddAdjustment:
+		y, err := resource.ParseQuantity(adj.Value)
+		if err != nil {
+			return fmt.Errorf("unable to parse value to resource.Quantity: %w", err)
+		}
+		m.Add(y)
+	case MulAdjustment:
+		y, err := strconv.Atoi(adj.Value)
+		if err != nil {
+			return fmt.Errorf("unable to parse value to int: %w", err)
+		}
+		m.Mul(int64(y))
+	}
+	return nil
+}
+
+type BroomRetryPolicy string
+
+const (
+	RetryAllowPolicy  BroomRetryPolicy = "Allow"
+	RetryForbidPolicy BroomRetryPolicy = "Forbid"
```

Comment:
Hmm, I think `retryPolicy: Allow, Forbid` is a bit confusing...

`restartPolicy: OnOOM, Never` like job.spec.restartPolicy. What do you think about this?

Reply:
I agree. Using `OnOOM` and `Never` makes so much sense!
I'm a little worried that "restart" might be misunderstood as another launch "in the same spec", just like Pod's `job.spec.restartPolicy`. Any thoughts on this?

Reply:
I'm starting to think "restart" is better after all. Will be fixed.